docs/github-action.md

```diff
@@ -0,0 +1,65 @@
+# GitHub Action
+
+If you want an easy way to test your repository spelling (or a subset of files)
+you can use the Typos Action! It is served from this repository, and can
+easily be used as follows:
+
+```yaml
+name: Test GitHub Action
+on: [pull_request]
+
+jobs:
+  run:
+    name: Spell Check with Typos
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Actions Repository
+      uses: actions/checkout@v2
+
+    - name: Test force pass with mistakes
+      uses: ./
+      with: 
+        files: ./src/file.rs
+        force_pass: true
+
+    - name: Test with no mistakes
+      run: echo "The quick brown fox jumped over the sleepy dog." > file.txt 
+    - uses: ./
+      with: 
+        files: ./file.txt
+        force_pass: false
+
+    - name: Test extend words
+      run: echo "The quick brown foxx jumped over the slepy dog." > file.txt 
+    - uses: ./
+      with: 
+        files: ./file.txt
+        force_pass: false
+        extend_words: "foxx,slepy"  
+
+    - name: Test extend works
+      run: echo "Ziggoman was the brother of Ziggogirl." > file.txt 
+    - uses: ./
+      with: 
+        files: ./file.txt
+        force_pass: false
+        extend_identifiers: "Ziggogirl,Ziggoman"
+```
+
+The "files" parameter is passed directly to the typos command, so you can
+add any additional parameters that you'd like there. Make sure when
+referencing relative file paths to use `./` (e.g., `./file.txt` instead of
+`file.txt`. You can also define identifiers and words to skip.
+
+## Variables
+
+| Name | Description | Required | Default |
+|------|-------------|----------|---------|
+| files| Files or patterns to check | false | If not defined, entire repository is checked |
+| extend_identifiers | Comma separated list of extend identifiers, like someone's name | false | not set |
+| extend_words | Comma separated list of extend words. | false | not set |
```

Comment:
The idea here is that these variables are written to a config file, but just in the action, so the user can add the logic of the check completely to the action without needing to add an extra file to the repository. That said, if the config file already exists it is honored (and these are ignored).

I'll add isolated and config!

Reply:
If people have need for specifying this config, they should be providing the config file.  This becomes a hinderance to the developer or contributors because how they run `typos` locally won't work with how it is run in CI.

Now, I could see us adding support for pulling the config from project files, e.g. `Cargo.toml` or `pyproject.toml`s metadata / tool tables.

Reply:
okay, we can just remove those extra arguments then. I'm not sure I'll use the action but I'm sure others will!